### PR TITLE
Gatsby cli repl test

### PR DESCRIPTION
**Pull request by blainekasten**
## Description

Adds basic tests for gatsby repl to ensure it starts a prompt. I couldn't figure out a way to send commands to the prompt, so this is at least a basic test to prove that the prompt starts

### CODE REVIEW

File: integration-tests/gatsby-cli/__tests__/repl.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    // 2. Wait for the build process to finish
+    await timeout(4)
+
+    // 3. Kill the repl
+    childProcess.kill()
+
+    // 4. Make assertions
+    const logs = getLogs()
+    logs.should.contain(`success open and validate gatsby-configs`)
```
Comment (freiksenet) [inline integration-tests/gatsby-cli/__tests__/repl.js:29]: Should this be tested? All those lines are kinda flaky and subject to change.

File: integration-tests/gatsby-cli/__tests__/repl.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  beforeAll(() => GatsbyCLI.from(cwd).invoke(`clean`))
+  afterAll(() => GatsbyCLI.from(cwd).invoke(`clean`))
+
+  it(`starts a gatsby site on port 8000`, async () => {
+    // 1. Start the `gatsby develop` command
+    const [childProcess, getLogs] = GatsbyCLI.from(cwd).invokeAsync(`repl`)
+
+    // 2. Wait for the build process to finish
+    await timeout(4)
```
Comment (freiksenet) [inline integration-tests/gatsby-cli/__tests__/repl.js:22]: Might fail because of that on slow CIs.

Comment (blainekasten) [reply integration-tests/gatsby-cli/__tests__/repl.js:22]: Yeah i realize that. I was just going to do the skateboard implementation and improve it later if it needed more stability. So far it's seemed to have worked.

File: integration-tests/gatsby-cli/__tests__/repl.js (hunk 1 above)
Comment (blainekasten) [reply integration-tests/gatsby-cli/__tests__/repl.js:29]: I think the point of this is to make sure the user experience is managed. So for now this is what's happening, and if it changes we should know. So I like having these tests for now.

**Pull request merged**